compiler/rustc_hir/src/definitions.rs

```diff
@@ -240,12 +244,9 @@ impl DefPath {
         for component in &self.data {
             s.extend(opt_delimiter);
             opt_delimiter = Some('-');
-            if component.disambiguator == 0 {
-                write!(s, "{}", component.data.as_symbol()).unwrap();
-            } else {
-                write!(s, "{}[{}]", component.data.as_symbol(), component.disambiguator).unwrap();
-            }
+            write!(s, "{}", component).unwrap();
```

Comment:
This also wants the verbose mode, as we don't want filenames to accidentally overlap.

Reply:
The implementation of `fmt::Display` for `DisambiguatedDefPathData` is already verbose so I think this is fine?

Reply:
Oops, I missed that part, I expected the opposite.